helix-db/src/helix_gateway/worker_pool/mod.rs

```diff
@@ -236,39 +236,51 @@ impl Worker {
             // Set thread local context, so we can access the io runtime
             let _io_guard = io_rt.enter();
 
-            // Simple loop for the writer - no parity needed since it's a single thread
+            // Use Selector to wait on both channels simultaneously.
+            // This is required because with a single writer thread, we can't use the
+            // parity trick that read workers use (which requires multiple threads).
```

Comment:
Correct, but continuation functions *must* be given priority otherwise it'll hang when another write request is sent before another write requiring continuation has finished. The selector must either not have fairness or explicitly poll continuation channel before this selector.

Test suite should be updated to catch the error of what I described above, since this will hang